apps/readiness_healthcheck.go

```diff
@@ -0,0 +1,134 @@
+package apps
+
+import (
+	"fmt"
+
+	. "github.com/cloudfoundry/cf-acceptance-tests/cats_suite_helpers"
+	"github.com/cloudfoundry/cf-acceptance-tests/helpers/app_helpers"
+	"github.com/cloudfoundry/cf-acceptance-tests/helpers/assets"
+	"github.com/cloudfoundry/cf-acceptance-tests/helpers/logs"
+	"github.com/cloudfoundry/cf-acceptance-tests/helpers/skip_messages"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gexec"
+
+	"github.com/cloudfoundry/cf-acceptance-tests/helpers/random_name"
+	"github.com/cloudfoundry/cf-test-helpers/v2/cf"
+	"github.com/cloudfoundry/cf-test-helpers/v2/helpers"
+	"github.com/cloudfoundry/cf-test-helpers/v2/workflowhelpers"
+)
+
+var _ = AppsDescribe("Readiness Healthcheck", func() {
+	var appName, proxyAppName string
+	var orgName string
+	var spaceName string
+	var readinessHealthCheckTimeout = "25s" // 20s route emitter sync loop + 2s hc interval + bonus
+
+	BeforeEach(func() {
+		if !Config.GetReadinessHealthChecksEnabled() {
+			Skip(skip_messages.SkipReadinessHealthChecksMessage)
+		}
+		appName = random_name.CATSRandomName("APP")
+		proxyAppName = random_name.CATSRandomName("APP")
+		orgName = TestSetup.RegularUserContext().Org
+		spaceName = TestSetup.RegularUserContext().Space
+	})
+
+	AfterEach(func() {
+		app_helpers.AppReport(appName)
+		app_helpers.AppReport(proxyAppName)
+		Eventually(cf.Cf("delete", appName, "-f")).Should(Exit(0))
+		Eventually(cf.Cf("delete", proxyAppName, "-f")).Should(Exit(0))
+	})
+
+	Describe("when the readiness healthcheck is set to http", func() {
+		FIt("registers the route only when the readiness check passes", func() {
+			By("pushing the app")
+			Expect(cf.Push(appName,
```

Comment:
You should use `cf.Cf("push",...)` instead. That way no changes are required to [cloudfoundry/cf-test-helpers](https://github.com/cloudfoundry/cf-test-helpers).

Reply:
💡 Good idea! @geofffranks you should do that :)

Reply:
Done